src/Compilers/CSharp/Portable/Symbols/Metadata/PE/PEParameterSymbol.cs

```diff
@@ -291,7 +291,7 @@ internal override bool HasMetadataConstantValue
         {
```

Comment:
It looks like ExplicitDefaultConstantValue has two options:

1. If the parameter is not optional then it returns null.
1. Otherwise return ImportConstantValue

ImportConstantValue either

1. Loads from ParamDefaultValue if HasDefault is selected
2. Loads a decimal or datetime

The default implementation of HasDefaultConstantValue asks if ExplicitDefaultConstantValue is null, decimal, or datetime and returns false in any of those cases. The implementation of ExplicitDefaultConstantValue only returns a non-null, non-datetime, or non-decimal value only if HasDefault is set.

So it seems like simply removing the override has the same logic and may be closer to the intended logic of the interface than overriding.